dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/scheduling/DefaultSchedulingManager.java

```diff
@@ -172,52 +173,48 @@ public void jobConfigurationFinished( JobConfiguration jobConfiguration )
     @Override
     public void scheduleJob( JobConfiguration jobConfiguration )
     {
-        if ( ifJobInSystemStop( jobConfiguration.getUid() ) )
-        {
-            JobInstance jobInstance = new DefaultJobInstance( this, messageService, leaderManager );
-
-            if ( jobConfiguration.getUid() != null && !futures.containsKey( jobConfiguration.getUid() ) )
-            {
-                log.info( String.format( "Scheduling job: %s", jobConfiguration ) );
-
-                ScheduledFuture<?> future = null;
-
-                if ( jobConfiguration.getJobType().isCronSchedulingType() )
-                {
-                    future = jobScheduler.schedule( () -> jobInstance.execute( jobConfiguration ),
-                        new CronTrigger( jobConfiguration.getCronExpression() ) );
-                }
-                else if ( jobConfiguration.getJobType().isFixedDelaySchedulingType() )
-                {
-                    future = jobScheduler.scheduleWithFixedDelay( () -> jobInstance.execute( jobConfiguration ),
-                        Instant.now().plusSeconds( DEFAULT_INITIAL_DELAY_S ),
-                        Duration.of( jobConfiguration.getDelay(), ChronoUnit.SECONDS ) );
-                }
-
-                futures.put( jobConfiguration.getUid(), future );
+        scheduleJob( jobConfiguration, this::scheduleCronOrDelayJob );
+    }
 
-                log.info( String.format( "Scheduled job: %s", jobConfiguration ) );
-            }
-        }
+    private ScheduledFuture<?> scheduleCronOrDelayJob( JobConfiguration jobConfiguration, JobInstance jobInstance )
+    {
+        log.info( String.format( "Scheduling job: %s", jobConfiguration ) );
+        SchedulingType type = jobConfiguration.getJobType().getSchedulingType();
+        ScheduledFuture<?> future = type == SchedulingType.CRON
```

Comment:
This is good. Just wondering, is readability suffering when we have such long lambdas/ternary statements, maybe make it slightly more old-school?

Reply:
I wasn't too happy about it either. Had tried a few variants. Enum with two values seems to throw off static analysis a bit too complaining about some solutions. Now looking at it it probably is a readable solution if both cases of the ternary get extracted to a helper method? WDYT?

Reply:
Extracting made this readable.